### PR TITLE
Adds csv names to the spec of installplan after unpacking bundles.

### PR DESCRIPTION
**Pull request by fantapsody**
**Description of the change:**

Adds the CSV name to `InstallPlanSpec.ClusterServiceVersionNames` after unpacking a bundle.

**Motivation for the change:**

Addressing https://github.com/operator-framework/operator-lifecycle-manager/issues/1347, especially the case I mentioned in the comment, that packages from a CatalogSource defined by an index image cannot be upgraded and reports error like `pre-existing CRD owners found for owned CRD(s) of dependent CSV`. The `ClusterServiceVersionNames` field in the Spec of InstallPlan is empty, although steps of kind `CustomResourceDefinition` exists in the Status.

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: pkg/controller/operators/catalog/operator.go [hunk 1]
```diff
@@ -1100,6 +1100,18 @@ func (o *Operator) unpackBundles(plan *v1alpha1.InstallPlan) (bool, *v1alpha1.In
 		// Add steps and remove resolved bundle lookup
 		out.Status.Plan = append(out.Status.Plan, steps...)
 		out.Status.BundleLookups = append(out.Status.BundleLookups[:i], out.Status.BundleLookups[i+1:]...)
+		// add csv name to the list
```
Comment (ecordell) [inline pkg/controller/operators/catalog/operator.go]: We should know these names before we unpack the bundles - they're returned as a part of the resolved set of operators from the resolver

File: pkg/controller/operators/catalog/operator.go [hunk 2]
```diff
@@ -1140,6 +1152,14 @@ func (o *Operator) syncInstallPlans(obj interface{}) (syncError error) {
 			return
 		}
 
+		if !reflect.DeepEqual(plan.Spec.ClusterServiceVersionNames, out.Spec.ClusterServiceVersionNames) {
```
Comment (ecordell) [inline pkg/controller/operators/catalog/operator.go]: Updating the spec here, followed by updating the status right below, will be a source of errors/flakes (i.e. we have to be sure to handle the case where updating the spec errors before updating the status).

**Comment by fantapsody** [comment]
Thanks for the fix, I will close this PR. @ecordell

**Pull request closed without merging**